contrib/endpoints/src/api_manager/request_handler.cc

```diff
@@ -79,6 +105,35 @@ void RequestHandler::AttemptIntermediateReport() {
 // Sends a report.
 void RequestHandler::Report(std::unique_ptr<Response> response,
                             std::function<void(void)> continuation) {
+  if (api_manager_->IsConfigLoadingInProgress()) {
+    if (pending_request_callback_exist_) {
+      continuation();
+      return;
+    }
+
+    std::shared_ptr<Response> copied_response(std::move(response));
+    api_manager_->AddPendingRequestCallback(
+        [copied_response, continuation, this](utils::Status status) {
+          if (status.ok()) {
+            InternalReport(
+                std::unique_ptr<Response>(std::move(copied_response.get())),
```

Comment:
passing a shared_ptr to unique_ptr is very dangerous.   unique_ptr may delete it and share_ptr may delete it too.   I suggest, pass raw pointer.